src/Reflection/ReflectionClass.php

```diff
@@ -236,11 +240,18 @@ private function createMethodsFromTrait(ReflectionMethod $method): array
     {
         $traitAliases     = $this->getTraitAliases();
         $traitPrecedences = $this->getTraitPrecedences();
+        $traitModifiers   = $this->getTraitModifiers();
 
         $methodAst = $method->getAst();
         assert($methodAst instanceof ClassMethod);
 
-        $methodHash   = $this->methodHash($method->getImplementingClass()->getName(), $method->getName());
+        $methodHash = $this->methodHash($method->getImplementingClass()->getName(), $method->getName());
+
+        if (isset($traitModifiers[$methodHash])) {
```

Comment:
Let's use `array_key_exists()` for these - `isset()` is just generally too risky for type-level regressions later on